compressor/offline/jinja2.py

```diff
@@ -84,7 +84,7 @@ def get_init_context(self, offline_context):
 
         # Allow offline context to override the globals.
         context = self.env.globals.copy()
-        context.update(offline_context)
+        context.update(flatten_context(offline_context))
```

Comment:
It's a relic of the first attempt to improve memory consumption, it's needed to be able to use a generator for the context function.
It may or may not be part of this PR, as you prefer. In the end, that's not how performance where improved, but it's nice that you can give a generator to compressor without errors.